### PR TITLE
sd-dhcp-client: introduce sd_dhcp_client_is_running()

### PR DESCRIPTION
**Pull request by yuwata**
Split out from #19271.

### CODE REVIEW

File: src/libsystemd-network/sd-dhcp-client.c [hunk 1]
```diff
@@ -2052,6 +2052,12 @@ int sd_dhcp_client_send_renew(sd_dhcp_client *client) {
         return client_initialize_time_events(client);
 }
 
+int sd_dhcp_client_is_running(sd_dhcp_client *client) {
+        assert_return(client, -EINVAL);
```
Comment (keszybz) [inline src/libsystemd-network/sd-dhcp-client.c]: Just return false in this case, like in 3bbb76f621?

File: src/libsystemd-network/sd-dhcp-client.c [hunk 2]
```diff
@@ -2052,6 +2052,12 @@ int sd_dhcp_client_send_renew(sd_dhcp_client *client) {
...
+
+        return client->state != DHCP_STATE_INIT;
```
Comment (keszybz) [inline src/libsystemd-network/sd-dhcp-client.c]: This returns true also for `DHCP_STATE_STOPPED`... On purpose?

File: src/systemd/sd-dhcp-client.h [hunk 3]
```diff
@@ -199,6 +199,7 @@ int sd_dhcp_client_set_fallback_lease_lifetime(
 int sd_dhcp_client_add_option(sd_dhcp_client *client, sd_dhcp_option *v);
 int sd_dhcp_client_add_vendor_option(sd_dhcp_client *client, sd_dhcp_option *v);
 
+int sd_dhcp_client_is_running(sd_dhcp_client *client);
```
Comment (keszybz) [inline src/systemd/sd-dhcp-client.h]: `const`?

**Review by keszybz — CHANGES_REQUESTED** [review]

**Comment by yuwata** [comment]
@keszybz Thank you for your review. All comments are addressed.

**Review by bluca — APPROVED** [review]
LGTM, but waiting for @keszybz since he left comments earlier

**Pull request merged**